pkg/util/mon/resource.go

```diff
@@ -35,7 +35,8 @@ func (m memoryResource) NewBudgetExceededError(
 ) error {
 	return pgerror.NewErrorf(
 		pgerror.CodeOutOfMemoryError,
-		"memory budget exceeded: %d bytes requested, %d currently allocated, %d bytes in budget",
+		`memory budget exceeded: %d bytes requested, %d currently allocated, %d bytes in budget
+try increasing memory budget with --max-sql-memory; for details see https://tinyurl.com/y8cctlna`,
 		requestedBytes,
 		reservedBytes,
 		budgetBytes,
```

Comment:
1) a hint is written using `SetHintf` not in the main error message
2) the response to such an error should not be to blindly increase the budget; perhaps there is another issue at hand. The hint must be more subtle.
3) don't rely on a 3rd-party service (tinyurl).



Try this:
```go
return pgerror.NewErrorf(
 		pgerror.CodeOutOfMemoryError,
		"memory budget exceeded: %d bytes requested, %d currently allocated, %d bytes in budget",
    requestedBytes, reservedBytes, budgetBytes,
    ).SetHintf("for more information, see: http://....")
```

and instead make a doc page which:

- explains the error
- explains that the default budget is 128MB, and should be configured in production deployments
- highlights that the adequate response to this error is:
  - to increase the budget if it's definitely too low (what "too low" means depends on the deployment, a good guidance should be: either the largest dataset they need for a single query, or 10 megabytes per connected client, whichever is higher)
  - if the budget is already adequate, investigate memory usage of the query and either rewrite the query or report an issue to Cockroach Labs

Reply:
Also be mindful that in most database deployments, the person who is administrating the database is very different from the person writing apps or using the apps.

The memory error will appear on the screen of end-users, not administrators. End users have no control over the database settings. Maybe the hint should be instead:

"this error may indicate that your database administrator must customize their database settings"

and have a prominent FAQ for administrators to peruse.

Reply:
I would add the FAQ and then link to it with `SetHintf` in this error message.